src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -0,0 +1,680 @@
+    {
+        if (this->mHandlers[i] == handler)
+        {
+            position = i;
+            break;
+        }
+    }
+
+    // Verify Handler was found
+    VerifyOrReturnValue(position < kMaxSceneHandlers, CHIP_NO_ERROR);
+
+    uint8_t nextPos = static_cast<uint8_t>(position + 1);
+    uint8_t moveNum = static_cast<uint8_t>(kMaxSceneHandlers - nextPos);
+
+    // TODO: Implement general array management methods
+    // Compress array after removal
+    memmove(&this->mHandlers[position], &this->mHandlers[nextPos], sizeof(this->mHandlers[position]) * moveNum);
+
+    this->mNumHandlers--;
+    // Clear last occupied position
+    this->mHandlers[mNumHandlers] = nullptr;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultSceneTableImpl::UnregisterAllHandler()
+{
+    for (uint8_t i = 0; i < this->mNumHandlers; i++)
+    {
+        ReturnErrorOnFailure(this->UnregisterHandler(this->mHandlers[0]));
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultSceneTableImpl::SceneSaveEFS(SceneTableEntry & scene)
+{
+    if (!this->HandlerListEmpty())
+    {
+        for (uint8_t i = 0; i < this->mNumHandlers; i++)
+        {
+            clusterId cArray[kMaxClusterPerScenes];
+            Span<clusterId> cSpan(cArray);
+            if (this->mHandlers[i] != nullptr)
+            {
+                this->mHandlers[i]->GetSupportedClusters(scene.mStorageId.mEndpointId, cSpan);
+                for (uint8_t j = 0; j < cSpan.size(); j++)
+                {
+                    ExtensionFieldsSet EFS;
+                    MutableByteSpan EFSSpan = MutableByteSpan(EFS.mBytesBuffer, kMaxFieldsPerCluster);
+
+                    EFS.mID = cArray[j];
+                    ReturnErrorOnFailure(this->mHandlers[i]->SerializeSave(scene.mStorageId.mEndpointId, EFS.mID, EFSSpan));
+                    EFS.mUsedBytes = (uint8_t) EFSSpan.size();
+                    ReturnErrorOnFailure(scene.mStorageData.mExtensionFieldSets.InsertFieldSet(EFS));
+                }
+            }
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultSceneTableImpl::SceneApplyEFS(FabricIndex fabric_index, const SceneStorageId & scene_id)
+{
+    FabricSceneData fabric(fabric_index);
+    SceneTableData scene(fabric_index);
+    ExtensionFieldsSet EFS;
+    TransitionTimeMs time;
+    clusterId cluster;
+
+    ReturnErrorOnFailure(fabric.Load(mStorage));
+    VerifyOrReturnError(fabric.Find(scene_id, scene.index) == CHIP_NO_ERROR, CHIP_ERROR_NOT_FOUND);
+    ReturnErrorOnFailure(scene.Load(mStorage));
+
+    if (!this->HandlerListEmpty())
+    {
+        for (uint8_t i = 0; i < scene.mStorageData.mExtensionFieldSets.GetFieldNum(); i++)
+        {
+            scene.mStorageData.mExtensionFieldSets.GetFieldSetAtPosition(EFS, i);
+            cluster = EFS.mID;
+            time    = scene.mStorageData.mSceneTransitionTimeSeconds * 1000 +
+                (scene.mStorageData.mTransitionTime100ms ? scene.mStorageData.mTransitionTime100ms * 10 : 0);
```

Comment:
So... I realize that the scene commands encode transition times like that, but why wouldn't we just store total ms in our storage all the time and make things simpler?  If the concern is storage space, right now we are storing two TLV things; one is a uint16_t and one is a uint8_t.  That uses either 6 or 7 bytes depending on whether the `uint16_t` actually fits in one byte (control + tag take 2 bytes per TLV item).

If we stored a single uint32_t millisecond count, then we would use at most 6 bytes.  And to get to 6 bytes we would need to have our time be at least 16,777 seconds, which would mean that the two-item thing we do now would have been 7 bytes anyway at that point.

Reply:
This was made so to be consistent with the spec. To be honest, I don't see the interest in having the value in tenth of a second rather than in milliseconds from the get go if we are going to store it in milliseconds and apply it in milliseconds. Will be changed.

Reply:
Now using milisecond in storage.

Reply:
This is a change that is not compatible with Zigbee version of Scenes. We have to maintain as much compatibility as possible, otherwise we don't have WG backing for these changes. Any changes that are beyond *clarifications* to content  in Matter 1.1 ballot text (and based on 1.0 provisional text) would require additional TSG round-trips with calls.

Reply:
How is this not compatible?  All of this is internal implementation details.